src/install/bin.zig

```diff
@@ -67,36 +67,39 @@ pub const Bin = extern struct {
     }
 
     pub fn clone(this: *const Bin, buf: []const u8, prev_external_strings: []const ExternalString, all_extern_strings: []ExternalString, extern_strings_slice: []ExternalString, comptime StringBuilder: type, builder: StringBuilder) Bin {
-        return switch (this.tag) {
-            .none => Bin{ .tag = .none, .value = .{ .none = {} } },
-            .file => Bin{
-                .tag = .file,
-                .value = .{ .file = builder.append(String, this.value.file.slice(buf)) },
+        var cloned = std.mem.zeroes(Bin);
```

Comment:
I think we can be a little safer here about zeroing if we instead create it via `@bitCast`

```suggestion
        var cloned_bytes = .{0} ** @byteSize(Bin);
        var cloned: Bin = @bitCast(cloned_bytes);
```

Reply:
need some help, I'm not familiar with `{0} **`. When I changed it to this

```zig
        const cloned_bytes = .{0} ** @sizeOf(Bin);
        var cloned: Bin = @bitCast(cloned_bytes);
```

compiled error:

```
src/install/bin.zig:74:36: error: cannot @bitCast from 'struct{comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0, comptime comptime_int = 0}'; struct does not have a guaranteed in-memory layout
        var cloned: Bin = @bitCast(cloned_bytes);
                                   ^~~~~~~~~~~~

```

And change to 

```zig
        const cloned_bytes: [@sizeOf(Bin)]u8 = .{0} ** @sizeOf(Bin);
        var cloned: Bin = @bitCast(cloned_bytes);
```

got


```
zig build-obj bun-debug Debug native-native-gnu.2.27: error: thread 2127830 panic: TODO implement readFromMemory for more types
Unable to dump stack trace: debug info stripped

```

Reply:
ah yeah it's 
```suggestion
const cloned_bytes = [@sizeOf(Bin)]u8 = [_]u8{0} ** @sizeOf(Bin);
var cloned: Bin = @bitCast(cloned_bytes);
```

Reply:
if that continues to fail then i suggest 

```zig
var cloned: Bin = std.mem.zeroes(Bin);
@memset(std.mem.asBytes(&cloned), 0);
```

Reply:
Fixed. `[_]u8{0} ** @sizeOf(Bin);` continues to fail with `TODO implement readFromMemory for more types`. I change it to `@memset`.